### PR TITLE
Fix tagging

### LINKED ISSUE #487: Make @tagging work
# Problem
In addition to the tagging issue reported on July 11 on https://github.com/near/neardevhub-widgets/issues/200, there are a couple of major usability issues with tagging / @mentioning specific people today:

## Issue 1: Description not showing selected person
**Current:** When** typing "@" plus a person's first few letters, such as "elli", I see the suggested list of people. When I click Elliot Braem, the preview shows their profile image and name. However, the description does not complete the username. So if I only type "@ellio" and then click on Elliot, the description markdown text still shows @ellio instead of showing @elliotbraem.near If I proceed with typing additional text after @ellio, the person is removed altogether from the preview, which defeats the purpose of the tagging functionality.

**Desired:** Similar to the behavior on near.org, when I select a person's name, their full username should populate within the description. So if I type @elli and select Elliot, I should see @elliotbraem.near in the description.

## Issue 2: Unable to select username when there is additional text in the description

**Current:** When I type a bunch of text in the description, if I then try to type "@" anywhere, the @ selection pop up does not show.

**Desired:** The selection popup should appear anytime I type @, regardless if there is already text. This should also apply if I @ mention a secondary, third, or nth name.

# Video
https://github.com/near/neardevhub-widgets/assets/110252255/9adf6b03-8948-4772-b3c0-5210d1ba1e62

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
Resolves #487

### CODE REVIEW

File: src/devhub/page/create.jsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const AutoComplete = styled.div`
 `;
 
 function textareaInputHandler(value) {
-  const showAccountAutocomplete = /@[\w][^\s]*$/.test(value);
+  const showAccountAutocomplete = /(^|\s)@/.test(value);
```
Comment (ailisp) [inline src/devhub/page/create.jsx:13]: Will this make autocomplete bar always showup, as `@` will always exist?

**Comment by ailisp** [comment]
Thanks @Megha-Dev-19 ! This seems like a right fix, but I'm not sure. Could you provide a preview link or a short screen recording to ensure two issues reported by @ori-near is fixed?

**Comment by Megha-Dev-19** [comment]
Hey @ailisp yeah sure, attaching the screen recording here.
also for preview I will have to deploy it on my account and then share the link right? or is there any other way?

https://github.com/near/neardevhub-widgets/assets/100185149/baf7bc2a-914e-422b-86de-7d7b69e24a0a

File: src/devhub/page/create.jsx (hunk 1 above)
Comment (Megha-Dev-19) [reply src/devhub/page/create.jsx:13]: no actually not
it works when there is a whitespace before @ or if it's in the beginning of the text

**Comment by Tguntenaar** [comment]
> Hey @ailisp yeah sure, attaching the screen recording here. also for preview I will have to deploy it on my account and then share the link right? or is there any other way?

Exactly for preview, you can deploy it on your account and then share the link in the PR

**Comment by Megha-Dev-19** [comment]
@Tguntenaar here is the preview link
https://near.org/megha19.near/widget/app?page=create

**Comment by ailisp** [comment]
@Megha-Dev-19 Thanks for your response, I tried myself. The issue 1 is fixed, but not the issue 2, see Ori's video in #487 from 00:40, when you try `@` in the middle of the post, not as the last word, the autocompletion still doesn't show up.

**Comment by Megha-Dev-19** [comment]
hey @ailisp you're right, I will check this

**Comment by ailisp** [comment]
Looks perfect now! Thank you!

**Review by ailisp — APPROVED** [review]

**Comment by ori-near** [comment]
Looks so good, thank you!

**Pull request merged**